src/rules/objectLiteralContextualTypeRule.ts

```diff
@@ -0,0 +1,91 @@
+/**
+ * @license
+ * Copyright 2017 Palantir Technologies, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { isArrayLiteralExpression, isObjectLiteralExpression, isPropertyAssignment } from "tsutils";
+import * as ts from "typescript";
+
+import * as Lint from "../index";
+
+export class Rule extends Lint.Rules.TypedRule {
+    /* tslint:disable:object-literal-sort-keys */
+    public static metadata: Lint.IRuleMetadata = {
+        ruleName: "object-literal-contextual-type",
+        description: "Requires that every object literal has a contextual type.",
+        rationale: Lint.Utils.dedent`
+            An object literal with no type does not have excess properties checked.
+
+            For example:
+
+                interface I { x: number; }
+                function f(): I {
+                    const res = { x: 0, y: 0 };
+                    return res;
```

Comment:
this error can be avoided so many ways that don't require a rather aggressive lint rule...

Reply:
How would it be avoided in general? I don't believe typescript's behavior with respect to object literals has changed much in the past year.
Often people will think they're getting a contextual type when they really aren't, e.g. a result from a callback:
```ts
interface Options { foo?: number; bar?: number }
const arr: Options[] = [1, 2, 3].map(n => ({ foo: n, baa: n }));
```
The error is fixed by changing to `(n): Options`, but I wouldn't want a lint rule to require all arrow functions to have explicit return types.
If there are better ways of avoiding this error I'd like to hear about them.